projects/extension/src/background/AppMediator.ts

```diff
@@ -61,19 +48,38 @@ export class AppMediator extends (EventEmitter as { new(): StateEmitter }) {
    */
   constructor(port: chrome.runtime.Port, manager: ConnectionManagerInterface) {
     super();
-    this.subscriptions = [];
-    this.requests = [];
     this.#appName = port.name.substr(0, port.name.indexOf('::'));
     this.#name = port.name;
     this.#port = port;
     this.#tabId = port.sender?.tab?.id;
     this.#url = port.sender?.url;
     this.#manager = manager;
     // Open listeners for the incoming rpc messages
-    this.#port.onMessage.addListener(this.#handleRpcRequest);
+    this.#port.onMessage.addListener(this.#handleMessage);
     this.#port.onDisconnect.addListener(() => { this.#handleDisconnect() });
   }
 
+  /** 
+   * associate parses the name of the network from the port name and associates
+   * the app with the smoldot client or sends an error and disconnects the port
```

Comment:
This is slightly misleading. Association should happen with the chain name and not with the smoldot client (there is only 1).